lib/gis/file_name.c

```diff
@@ -27,8 +27,29 @@ static void append_char(char*, char);
   If <i>name</i> is of the form "nnn@ppp" then path is set as if name
   had been "nnn" and mapset had been "ppp" (mapset parameter itself is
   ignored in this case).
-  
-  \param[out] path buffer to hold resultant full path to file
+
+  Paths to files are in form:
+  /home/user/GISDBASE/LOCATION/mapset/element/name
```

Comment:
This might be misleading. GISDBASE var contains /home/user or generally anything else. Some alternatives:

```
mapset/element/name
.../mapset/element/name
/path/to/mapset/element/name
/path/to/LOCATION/mapset/element/name
.../location/mapset/element/name
$GISDBASE/$LOCATION/mapset/element/name
$GISDBASE/$LOCATION_NAME/mapset/element/name
```

I think the intention here was to hide the concrete path, so what path it is is an implementation detail which the user should not depend on, so perhaps you can say something like _Paths to files are currently in form:_ One use case for the change of implementation would be changing mapset structure (to raster/elevation/fcell misc-like layout), but supporting old mapset structure too. "open old" would then check both possible paths while "open new" would use the new style and delete the files in the old structure if they exist.

Reply:
I went with /full/path/to/mapset/element/name